Dynamic Forms: textarea and checkbox form control support

dmhuebner opened the pull request:
This PR adds support for a dynamic textarea component to the Form Generator.

dmhuebner commented:
It also adds support for a dynamic checkbox component.

nayfin commented on projects/tft-library/src/lib/dynamic-form/field-container/field-container.component.scss:
```diff
@@ -1,3 +1,7 @@
-}
\ No newline at end of file
+}
+
```
what are your thoughts on using flex-direction to accomplish? Which do you think will be more versatile?

nayfin replied:
That question may not make sense... 
I'm having trouble remembering why this was needed in the first place

dmhuebner replied:
Hey! Sorry for the delay. Yeah that's a good point. I opted to use the display property for maximum cross browser support, but we could totally switch it to using flexbox properties. It might be slightly nicer.

nayfin commented on projects/tft-library/src/lib/dynamic-form/field-container/field-container.component.ts:
```diff
@@ -15,6 +15,10 @@ export class FieldContainerComponent implements OnInit, OnDestroy {
   // the parent formGroup
   @Input() group: FormGroup;
+  // boolean whether to show label in container or not
+  @Input() showLabel: boolean = true;
```
Does this address a use case I am missing? I kind of like having it where if there is a label passed in we use, if there's not we hide the element. It minimizes configuration.

dmhuebner replied:
Yeah, so the situation I found had to do with the checkbox component. I set up the checkbox component to use the label property on the config for the checkbox text/option. Because the FieldContainer was also using the label, the checkbox text/option was displaying twice. I think a better way to do it would be to not mess with a showLabel flag in FieldContainer, and change the checkbox implementation to use something different from config.label. Maybe something like config.option or config.text makes more sense. I'll make those changes and see what you think.

dmhuebner replied:
changes made!

nayfin commented on projects/tft-library/src/lib/dynamic-form/form-checkbox/form-checkbox.component.html:
```diff
@@ -0,0 +1,14 @@
+<tft-field-container
```
I wonder if we should worry about vertical/horizontal lists of checkboxes or just let the multiselect option in the select boxes handle that scenario. Not important for this PR. Just something to think on

nayfin commented on projects/tft-library/src/lib/dynamic-form/form-checkbox/form-checkbox.component.ts:
```diff
@@ -0,0 +1,26 @@
+
+  ngOnInit() {
+    this.labelPosition = this.config.labelPosition || 'after';
+    this.inline = this.config.inline || false;
```
I don't like setting defaults like this, but I can't think of another way to get around it...

dmhuebner replied:
I hear ya.. Maybe we can brainstorm another way?

nayfin commented:
The only change I'd like to see before merging is changing `defaultRows` to `rows` if we can.

dmhuebner commented:
Thanks for reviewing!! I went ahead and changed `defaultRows` to `rows` and refactored the checkbox component template/config to use `.text` instead of `.label` to show the checkbox text which allowed me to remove the `showLabel` flag on the `FieldContainerComponent`. Let me know if you find anything else!
